### PR TITLE
Generate schema for `isograph.config.json`

### PR DESCRIPTION
**Pull request by PatrykWalach**

### CODE REVIEW

File: libs/isograph-compiler/isograph-config-schema.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "ConfigFile",
+  "type": "object",
+  "required": [
+    "project_root",
+    "schema"
+  ],
+  "properties": {
+    "$schema": {
```
Comment (rbalicki2) [inline libs/isograph-compiler/isograph-config-schema.json:10]: lol is this a typo or is schema a special field?

File: crates/isograph_config/src/compilation_options.rs [hunk 2]
```diff
@@ -95,7 +101,7 @@ struct ConfigFile {
     pub schema_extensions: Vec<PathBuf>,
 
     /// Various that are of lesser importance
```
Comment (rbalicki2) [inline crates/isograph_config/src/compilation_options.rs:103]: nit: `Various options of less importance`

**Review by rbalicki2 — CHANGES_REQUESTED** [review]
Yes, yes, yes. Love it. 1000%. Let's rebase on main and I'll re-review

File: libs/isograph-compiler/isograph-config-schema.json (hunk 1 above)
Comment (PatrykWalach) [reply libs/isograph-compiler/isograph-config-schema.json:10]: Yes, this is intentional

File: crates/graphql_artifact_generation/src/format_parameter_type.rs [hunk 3]
```diff
@@ -75,7 +75,7 @@ fn format_server_field_type(
 fn format_field_definition(
     schema: &ValidatedSchema,
     name: &SelectableFieldName,
-    type_: &FieldType<ServerFieldId, ClientType<ClientFieldId>>,
+    type_: &FieldType<ServerFieldId, ClientType<ClientFieldId, ClientPointerId>>,
```
Comment (rbalicki2) [inline crates/graphql_artifact_generation/src/format_parameter_type.rs:78]: sorry I meant to rebase just the commits that relate to generating the schema, i.e. we shouldn't have any changes related to ☝️

**Pull request closed without merging**